### ISSUE TITLE
lipo error when build against engine directly in xcode

### ISSUE THREAD

**Issue by xster** [flutter/flutter#8049]
Follow https://github.com/flutter/engine/blob/master/CONTRIBUTING.md 

Everything works. But if building inside xcode, the lipo step can't find the input file

```
/bin/sh -c /Users/xster/Library/Developer/Xcode/DerivedData/Runner-fskhddlawlwosafmbeulxpukdbmp/Build/Intermediates/Runner.build/Debug-iphoneos/Runner.build/Script-3B06AD1E1E4923F5004D2608.sh

error: lipo: no input files specified
fatal error: lipo: Usage: lipo [input_file] ... [-arch <arch_type> input_file] ... [-info] [-detailed_info] [-output output_file] [-create] [-arch_blank <arch_type>] [-thin <arch_type>] [-remove <arch_type>] ... [-extract <arch_type>] ... [-extract_family <arch_type>] ... [-verify_arch <arch_type> ...] [-replace <arch_type> <file_name>] ...
cp: /Users/xster/Library/Developer/Xcode/DerivedData/Runner-fskhddlawlwosafmbeulxpukdbmp/Build/Products/Debug-iphoneos/Runner.app/Frameworks/Flutter.framework/Flutter_merged: No such file or directory
Command /bin/sh emitted errors but did not return a nonzero exit code to indicate failure
```

It still runs and everything. It just omitted the lipo step. Probably some environment variable sharing issue between shell and xcode.

**Comment by cbracken** [comment]
Which version of flutter are you sync'ed to? Repro steps?

**Comment by xster** [comment]
More or less the same as the other one. Build iOS device or simulator. flutter create a project. flutter run works without complaining. Go into xcode. Running directly works too but prints a bunch of errors.

**Comment by cbracken** [comment]
Fixed by #8251
